awx/ui/test/e2e/tests/test-form-error-handling.js

```diff
@@ -0,0 +1,202 @@
+                data = { jt, jt2 };
+                done();
+            });
+        client
+            .login()
+            .waitForAngular()
+            .resizeWindow(1200, 1000)
+            .useXpath()
+            .findThenClick(templatesNavTab)
+            .findThenClick('//*[@id="button-add"]')
+            .findThenClick('//a[@ui-sref="templates.addJobTemplate"]');
+    },
+    'Test max character limit when creating a job template': client => {
+        client
+            .waitForElementVisible('//input[@id="job_template_name"]')
+            .setValue(
+                '//input[@id="job_template_name"]',
+                ['a'.repeat(513), client.Keys.ENTER]
+            )
+            .setValue(
+                '//input[@name="inventory_name"]',
+                ['test-form-error-handling-inventory', client.Keys.ENTER]
+            )
+            .setValue(
+                '//input[@name="project_name"]',
+                ['test-form-error-handling-project', client.Keys.ENTER]
+            )
+
+            // clicked twice to make the element an active field
+            .findThenClick('//*[@id="select2-playbook-select-container"]')
+            .findThenClick('//*[@id="select2-playbook-select-container"]')
+            .findThenClick('//li[text()="hello_world.yml"]')
+            .findThenClick('//*[@id="job_template_save_btn"]')
+            .findThenClick('//*[@id="alert_ok_btn"]');
+
+        client.expect.element('//div[@id="job_template_name_group"]' +
+            '//div[@id="job_template-name-api-error"]').to.be.visible.before(3000);
+    },
+
+    'Test duplicate template name handling when creating a job template': client => {
+        client
+            .waitForElementNotVisible('//*[@id="alert_ok_btn"]')
+            .waitForElementVisible('//div[contains(@class, "Form-title")]')
+            .clearValue('//input[@id="job_template_name"]')
+            .setValue(
+                '//input[@id="job_template_name"]',
+                ['test-form-error-handling-job-template', client.Keys.ENTER]
+            )
+            .findThenClick('//*[@id="job_template_save_btn"]')
+            .findThenClick('//*[@id="alert_ok_btn"]');
+    },
+
+    'Test incorrect format when creating a job template': client => {
+        client
+            .waitForElementNotVisible('//*[@id="alert_ok_btn"]')
+            .waitForElementVisible('//div[contains(@class, "Form-title")]')
+            .clearValue('//input[@name="inventory_name"]')
+            .setValue(
+                '//input[@name="inventory_name"]',
+                ['invalid inventory name', client.Keys.ENTER]
+            )
+            .clearValue('//input[@name="project_name"]')
+            .setValue(
+                '//input[@name="project_name"]',
+                ['invalid project', client.Keys.ENTER]
+            );
+
+        client.expect.element('//*[@id="job_template-inventory-notfound-error"]')
+            .to.be.visible.after(3000);
```

Comment:
we might want to consider using the explicit wait parameter if we find ourselves building this logic on many testsuites